src/MongoDB.Driver/Linq/Processors/Pipeline/MethodCallBinders/SelectManyBinder.cs

```diff
@@ -117,5 +117,15 @@ private static IFieldExpression GetCollectionField(Expression collectionSelector
 
             return null;
         }
+
+        private FieldExpression PrepareFieldExpression(IFieldExpression collectionField, IBsonSerializer serializer)
+        {
+            // Determine whether we generate a query based on a document itself (when a `Document` type is `DocumentExpression`)
+            // or the result of previous steps (the `Document` type is `FieldExpression`).
+            var fieldExpression = collectionField.Document as FieldExpression; 
+            return fieldExpression == null
+                ? new FieldExpression(collectionField.FieldName, serializer)
+                : new FieldExpression(new FieldExpression(fieldExpression.FieldName, serializer), collectionField.FieldName, serializer);
```

Comment:
Good question. Parent `FieldExpression` had the same serializator as the expression from the current level. However, It didn't really matter, since we didn't deserialize a parent expression value at all.
Anyway, I've fixed this issue in my last comment.